repos/spack_repo/builtin/packages/gettext/package.py

```diff
@@ -166,3 +169,11 @@ def libs(self):
             shared=True if not shared_variant else shared_variant.value,
         )
         return libs
+
+    # gettext doesn't produce a pkg-config
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        env.append_flags(
+            "LDFLAGS",
+            "-L{0} ".format(dependent_spec["gettext"].prefix.lib)
+            + dependent_spec["gettext"].libs.link_flags,
+        )
```

Comment:
I'm not sure about these, should we really unconditionally link against these libraries if they are a dependency?
Also, should these be limited to `self.spec.satisfies("~shared")`? Doesn't Spack's compiler wrapper add `-L...` already?